app/models/course/condition/achievement.rb

```diff
@@ -32,13 +33,16 @@ def achievement_conditions(achievement)
 
     # Workaround, pending the squeel bugfix (activerecord-hackery/squeel#390) that will allow
     # allow the above query to work without #reload
-    Course::Achievement.joins(
-      "INNER JOIN
+    Course::Achievement.joins(<<-SQL)
+      INNER JOIN
         (SELECT cca.achievement_id
           FROM course_condition_achievements cca INNER JOIN course_conditions cc
             ON cc.actable_type = 'Course::Condition::Achievement' AND cc.actable_id = cca.id
-            WHERE cc.conditional_id = #{achievement.id}) ids
```

Comment:
This query gives false validation when 2 different conditionals with the same id have the same condition e.g. when achievement 2 requires achievement 1 and assessment 2 also requires achievement 1. This validation fails (while it's not supposed to). The test catches this error.